maven35-agent/pom.xml

```diff
@@ -24,12 +24,48 @@
     <dependency>
       <groupId>org.codehaus.plexus</groupId>
       <artifactId>plexus-classworlds</artifactId>
+      <version>2.5.2</version>
     </dependency>
     <dependency>
       <groupId>org.eclipse.sisu</groupId>
       <artifactId>org.eclipse.sisu.plexus</artifactId>
-      <version>0.0.0.M2a</version>
+      <version>0.3.3</version>
     </dependency>
+    
+   <!-- Upper bounds from org.apache.maven:maven-embedder:3.5.0 -->
```

Comment:
The only use case we care to support is from `maven-plugin`, so I think we can freely change things if it makes maintaining that plugin easier in any way.